subprocess.h

```diff
@@ -118,14 +118,17 @@ subprocess_weak int subprocess_create(const char *const command_line[],
 /// for a child process (each element of the form FOO=BAR). The last element
 /// must be NULL to signify the end of the array.
 /// @param out_process The newly created process.
+/// @param process_cwd The CWD of the newly created process. If null, will be 
+/// the same as the parent process.
 /// @return On success zero is returned.
 ///
 /// If `options` contains `subprocess_option_inherit_environment`, then
 /// `environment` must be NULL.
 subprocess_weak int
 subprocess_create_ex(const char *const command_line[], int options,
                      const char *const environment[],
-                     struct subprocess_s *const out_process);
+                     struct subprocess_s *const out_process,
```

Comment:
`out_process` should be the last argument of the function (I think you did this just so that you didn't mess up existing users of the function, but if we cared about that we'd just have to introduce `subprocess_create_ex2` and cry a bit inside!). In general I'm ok with `_ex` functions having their signatures change, but less so with the base functions.